### DISCUSSION TITLE
Limiting which repo member can trigger an action

### DISCUSSION THREAD

**Discussion by jschuur** [community/community#26960] [Actions]
<p>I’m planning a build/deploy process that will trigger deploys to a test/staging/production environment based on the branch pushed to, but I want to make sure only some people with access to the repo can push to production.</p>
<p>I think I can just look at GITHUB_ACTOR and prevent an accidental deploy to the wrong branch based on that, but am I correct that there’s not much I can do to completely lock that down, since anyone with repo access, could just modify the Action config to remove any checks I add to it?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/jschuur">@jschuur</a> ,</p>
<p>You can set the <a href="https://help.github.com/en/github/administering-a-repository/enabling-branch-restrictions" rel="nofollow noopener"><strong>Branch protection rules</strong></a> for the <strong>production</strong> branch. there is an option " <strong>Restrict who can push to matching branches</strong>" can be used to specify people, teams or apps allowed to push to matching branches.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/34b878f38bc630805c2bcee11aec894738359f07.png" data-download-href="/uploads/short-url/7wo3EpnSjmpmXVDwwSTJsHdhAkT.png?dl=1" title="add-rule.PNG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/46307996/181100976-871b54ea-fb96-4beb-9685-b7171c06662c.png" alt="add-rule.PNG" data-base62-sha1="7wo3EpnSjmpmXVDwwSTJsHdhAkT" width="690" height="373" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/34b878f38bc630805c2bcee11aec894738359f07_2_690x373.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/34b878f38bc630805c2bcee11aec894738359f07_2_1035x559.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/3/34b878f38bc630805c2bcee11aec894738359f07.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/34b878f38bc630805c2bcee11aec894738359f07_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">add-rule.PNG</span><span class="informations">1233×668 53.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jschuur** [comment]
<p>Fantastic! This is perfect, thanks!</p>
